### ISSUE TITLE
Download failing without asking for authentication

### ISSUE THREAD

**Issue by scooterman** [github/git-lfs#1047]
Hello. I have two boxes, one with a windows 10 and one with a linux (fedora 23). On windows I'm being able to download the files from git-lfs sucessfully but not on linux. Here are the logs for GIT_TRACE on both of them:
windows:
http://pastebin.com/raw/Z4LVnzVR
linux:
http://pastebin.com/NqARdUny

On linux I've tried to set the cache and store behavior assigining credentials to ~/.git-credentials without success.

**Comment by technoweenie** [comment]
What does `ssh git@gitlab.com git-lfs-authenticate deonn/rebellion.git download` return?

**Comment by scooterman** [comment]
hi @technoweenie, thanks for your time. 
on both windows and linux the ssh returns:

> Enter passphrase for key '/c/Users/.../.ssh/id_rsa':
> /opt/gitlab/embedded/service/gitlab-shell/lib/gitlab_shell.rb:142:in `exec': No such file or directory - git-lfs-authenticate (Errno::ENOENT)
>         from /opt/gitlab/embedded/service/gitlab-shell/lib/gitlab_shell.rb:142:in `exec_cmd'
>         from /opt/gitlab/embedded/service/gitlab-shell/lib/gitlab_shell.rb:117:in `process_cmd'
>         from /opt/gitlab/embedded/service/gitlab-shell/lib/gitlab_shell.rb:34:in `exec'
>         from /opt/gitlab/embedded/service/gitlab-shell/bin/gitlab-shell:24:in `<main>'

**Comment by technoweenie** [comment]
Ah, looks like an issue with your server then.

**Comment by scooterman** [comment]
The thing is, on windows this works as spected (like the first pastebin). FYI this is on gitlab.com.

**Comment by scooterman** [comment]
It seems to me that since the auth is basic it should try directly over https?

**Comment by technoweenie** [comment]
There's no automatic cross-platform way to convert an SSH url into the appropriate HTTPS url. You can either setup your repo with an HTTPS remote, or you can set up a custom lfs endpoint. See https://github.com/github/git-lfs/blob/master/docs/spec.md#the-server for details.

**Comment by scooterman** [comment]
Thanks. But then how this is supposed to be working on my windows machine and not on my linux one?

**Comment by technoweenie** [comment]
What's in `/home/victor/dev/deonn/rebellion/rebellion/.git/lfs/objects/logs/20160225T095036.861416028.log`?

**Comment by scooterman** [comment]
Just a quick heads up: looking at the link you've showed, I've changed the original config line that was this:

> [lfs "https://gitlab.com/repo/repo.git/info/lfs"]
> 	access = basic

to this:

> [lfs]
>         url="https://gitlab.com/repo/repo.git/info/lfs"
>         access = basic

made the download to work, but checking the config file again it seems it duplicated the [lfs] configuration

> [lfs]
>         url="https://gitlab.com/repo/repo.git/info/lfs"
>         access = basic
> [lfs "https://gitlab.com/repo/repo.git/info/lfs"]
>         access = basic

this is the content of the file you requested:

> 
> git-lfs/1.1.1 (GitHub; linux amd64; go 1.5.3; git 7de0397)
> git version 2.5.0
> 
> $ git-lfs smudge Android/src/main/res/drawable-hdpi/icon.png
> Error downloading object: Android/src/main/res/drawable-hdpi/icon.png (8c0f31da4588e55bb156459716df55bde6a2780a7e274477e48594a7391b4128)
> 
> exit status 1
> goroutine 1 [running]:
> github.com/github/git-lfs/lfs.Stack(0x0, 0x0, 0x0)
> 	/Users/rick/go/src/github.com/github/git-lfs/lfs/errors.go:557 +0x80
> github.com/github/git-lfs/commands.logPanicToWriter(0x7f27b96861e8, 0xc82015a008, 0x7f27b96422e0, 0xc8201580f0)
> 	/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:184 +0xf7f
> github.com/github/git-lfs/commands.logPanic(0x7f27b96422e0, 0xc8201580f0, 0x0, 0x0)
> 	/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:148 +0x421
> github.com/github/git-lfs/commands.handlePanic(0x7f27b96422e0, 0xc8201580f0, 0x0, 0x0)
> 	/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:123 +0x4e
> github.com/github/git-lfs/commands.LoggedError(0x7f27b96422e0, 0xc8201580f0, 0x96a3a0, 0x21, 0xc82014fc98, 0x2, 0x2)
> 	/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:73 +0x82
> github.com/github/git-lfs/commands.smudgeCommand(0xb5cfa0, 0xc8200eaa80, 0x1, 0x1)
> 	/Users/rick/go/src/github.com/github/git-lfs/commands/command_smudge.go:77 +0xd89
> github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra.(*Command).execute(0xb5cfa0, 0xc8200ea9d0, 0x1, 0x1, 0x0, 0x0)
> 	/Users/rick/go/src/github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra/command.go:477 +0x403
> github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra.(*Command).Execute(0xb5dfe0, 0x0, 0x0)
> 	/Users/rick/go/src/github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra/command.go:551 +0x46a
> github.com/github/git-lfs/commands.Run()
> 	/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:88 +0x23
> main.main()
> 	/Users/rick/go/src/github.com/github/git-lfs/git-lfs.go:34 +0x12e
> 
> ENV:
> LocalWorkingDir=/home/victor/dev/deonn/rebellion/rebellion
> LocalGitDir=/home/victor/dev/deonn/rebellion/rebellion/.git
> LocalGitStorageDir=/home/victor/dev/deonn/rebellion/rebellion/.git
> LocalMediaDir=/home/victor/dev/deonn/rebellion/rebellion/.git/lfs/objects
> TempDir=/home/victor/dev/deonn/rebellion/rebellion/.git/lfs/tmp
> ConcurrentTransfers=3
> BatchTransfer=true
> GIT_DIR=.git
> GIT_TRACE=1

**Comment by technoweenie** [comment]
> it made the download to work, but checking the config file again it seems it duplicated the [lfs] configuration

Not really. LFS is setting `lfs."https://gitlab.com/repo/repo.git/info/lfs".access`, and you changed it to `lfs.access`. LFS uses `git config` to manage the config files.

What does `git lfs env` look like? Do you have any LFS extensions setup? That looks like the only way that the smudge command could return a generic "exit status 1" error.

**Comment by scooterman** [comment]
here is the output of git lfs env:

`git-lfs/1.1.1 (GitHub; linux amd64; go 1.5.3; git 7de0397)
git version 2.5.0

Endpoint=https://gitlab.com/deonn/rebellion.git/info/lfs (auth=basic)
LocalWorkingDir=/home/victor/dev/deonn/rebellion
LocalGitDir=/home/victor/dev/deonn/rebellion/.git
LocalGitStorageDir=/home/victor/dev/deonn/rebellion/.git
LocalMediaDir=/home/victor/dev/deonn/rebellion/.git/lfs/objects
TempDir=/home/victor/dev/deonn/rebellion/.git/lfs/tmp
ConcurrentTransfers=3
BatchTransfer=true
git config filter.lfs.smudge = "git-lfs smudge %f"
git config filter.lfs.clean = "git-lfs clean %f"
`

**Comment by technoweenie** [comment]
What about `git lfs ext list`? If that doesn't spit anything useful out, I'm stuck (unless you want to try a custom build with lots of trace messages).

**Comment by scooterman** [comment]
The command returned nothing. I can run the trace version, no problem at all.

**Comment by sinbad** [comment]
I wonder if you've got 2 different versions of git-lfs there. I notice that in both outputs you have these lines:
```
trace git-lfs: ssh: git@gitlab.com git-lfs-authenticate deonn/rebellion.git download
trace git-lfs: ssh: download attempted with git@gitlab.com.  Error: exit status 1
```
Notice the error exit code at the end. The difference is that on Windows, it carries on anyway whilst on Linux it (correctly) fails out because of the SSH return code. I think that your Linux box is running git-lfs v1.1.1, and your Windows box is running git-lfs v1.1.0, because of this bugfix: https://github.com/github/git-lfs/pull/909 . Falling through is a bug that was fixed in 1.1.1.  

I think that the URL that's constructed in the incorrect fallthrough when the SSH call fails just *happens* to work on gitlab. This is still a bug in gitlab's SSH support if so, you just happen to be randomly benefitting from a bug that in your case is bypassing a genuine SSH failure. In other cases the URL that's constructed is completely wrong and it would miss cases of SSH key auth failure, hence the bugfix.

**Comment by sinbad** [comment]
TL;DR: use the https clone url since gitlab don't appear to be supporting SSH+LFS but you're lucking out with git-lfs v1.1.0 on Windows.

**Comment by scooterman** [comment]
@sinbad that's exactly the case. Thank you and @technoweenie for the support. Will get in touch with gitlab guys.

**Comment by JonasT** [comment]
The respective bug report seems to be here: https://gitlab.com/gitlab-org/gitlab-ce/issues/13541
Is there a known workaround? Should I just use 1.1.0 for now?

**Comment by technoweenie** [comment]
@JonasT: 

> use the https clone url

https://github.com/github/git-lfs/issues/1047#issuecomment-190283659
